sys-fs/mergerfs/mergerfs-2.24.2.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit toolchain-funcs autotools
+
+DESCRIPTION="A featureful FUSE union filesystem across multiple drives"
+HOMEPAGE="https://github.com/trapexit/mergerfs"
+# tarball under download/ has prebuilt ChangeLog and AUTHORS,
+# and more importantly version.hpp,
+# so use it instead of git tag tarball
+SRC_URI="https://github.com/trapexit/mergerfs/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="ISC"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="+xattr"
+# mergerfs bundles its own _patched_ copy of libfuse.
+# No easy way around it yet. :(
+#IUSE="${IUSE} +external-fuse"
+
+DEPEND="xattr? ( sys-apps/attr )"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-Makefile-don-t-touch-LDFLAGS.patch
+)
+
+DOCS=(
+	AUTHORS
+	ChangeLog
+	README.md
+)
+
+domake() {
+	local makeargs=( "$@" )
+	local makevars=(
+		GIT=''
+		MKDIR="$(type -P mkdir)"
```

Comment:
Are all those overrides (up to `INSTALL`) really necessary?

Reply:
Mostly yes.
I've double-checked and will drop a couple of them.
* `GIT` -- will drop
* `MKDIR` -- used during `make all` and `make install`
* `TOUCH` -- used during `make all`
* `CP` -- used during `make install`
* `RM` -- will drop
* `LN` -- used during `make install`
* `INSTALL` -- used during `make install`

Reply:
I've just looked at the Makefile and it's so horrible I can't find appropriate words to describe it. You need to DEPEND on `sys-apps/which` btw. Not that using `which` makes any sense (or `type -P` either, why would anyone need full paths there?!).

Reply:
No, I override these variables exactly to avoid `BDEPEND` on `which`.
Yeah, fair enough, I could have omitted `type -P` and used the commands without full paths.